PublicResolver.lll

```diff
@@ -0,0 +1,246 @@
+  ;; @param input A 32-byte number.
+
+  (def 'shift-left (input)
+    (mul input (exp 2 224)))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Shifts the leftmost 4 bytes of a 32-byte number right by 28 bytes.
+  ;; @param input A 32-byte number.
+
+  (def 'shift-right (input)
+    (div input (exp 2 224)))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Determines whether the supplied function ID matches a known
+  ;;         function hash and executes <code-body> if so.
+  ;; @dev The function ID is in the leftmost four bytes of the call data.
+  ;; @param function-hash The four-byte hash of a known function signature.
+  ;; @param code-body The code to run in the case of a match.
+
+  (def 'function (function-hash code-body)
+    (when (= (shift-right (calldataload 0x00)) function-hash)
+      code-body))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Hashes a given node to determine its actual storage location.
+  ;; @param node Hash this node address.
+
+  (def 'hash-location (node)
+    (seq
+      (mstore scratch-one node)
+      (sha3 scratch-one 32)))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Looks up the address for the node passed in.
+  ;; @param node Get the address for this node.
+
+  (def 'get-address (node)
+    (sload (hash-location node)))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Stores the address for the node passed in.
+  ;; @param node Set the address for this node.
+  ;; @param addr The address to set.
+
+  (def 'set-address (node addr)
+    (sstore (hash-location node) addr))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Looks up the content hash for the node passed in.
+  ;; @param node Get the content hash for this node.
+
+  (def 'get-hash (node)
+    (sload (+ (hash-location node) 1)))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Stores the content hash for the node passed in.
+  ;; @param node Set the content hash for this node.
+  ;; @param hash The content hash to set.
+
+  (def 'set-hash (node hash)
+    (sstore (+ (hash-location node) 1) hash))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Checks that the caller is the node owner.
+  ;; @param node Check the owner of this node.
+
+  (def 'only-owner (node)
+    (seq
+
+      ;; Call ens.owner(node).
+      (mstore (+ call-data 0x00) (shift-left get-node-owner))
+      (mstore (+ call-data 0x04) node)
+      (call (- (gas) 1000) 0 @@ens-registry call-data 36 return-data 32)
+
+      ;; If the call failed or the caller isn't the current node owner,
+      ;; throw an exception.
+      (when (|| (= @return-code false) (!= @return-data (caller)))
```

Comment:
I think you missed a [bug fix](https://github.com/ethereum/ens/pull/27/commits/598de2b735a421dc610196a2467bf37fcb7e8fbf) that was committed after the code you're looking at. In that code you'll see
```
(mstore return-code (call (- (gas) 1000) 0 @@ens-registry
```